test/uk/gov/hmrc/cardpaymentfrontend/controllers/CheckYourAnswersControllerSpec.scala

```diff
@@ -243,73 +286,105 @@ class CheckYourAnswersControllerSpec extends ItSpec {
+      assertRow(referenceRow, "Cyfeirnod Unigryw y Trethdalwr (UTR)", "1234567895K", None, None)
+    }
+
+    "[BtaSa] should render the payment date row correctly" in {
+      PayApiStub.stubForFindBySessionId2xx(TestJourneys.BtaSa.testBtaSaJourneyUpdatedWithRefAndAmount)
+      val result = systemUnderTest.renderPage(fakeRequest())
+      val document = Jsoup.parse(contentAsString(result))
+      val paymentDateRow = document.select(".govuk-summary-list__row").asScala.toList(0)
+      assertRow(paymentDateRow, "Payment date", "Today", Some("Change"), Some("some-link-to-pay-frontend"))
+    }
+
+    "[BtaSa] should render the payment date row correctly in Welsh" in {
+      PayApiStub.stubForFindBySessionId2xx(TestJourneys.BtaSa.testBtaSaJourneyUpdatedWithRefAndAmount)
+      val result = systemUnderTest.renderPage(fakeRequestWelsh())
+      val document = Jsoup.parse(contentAsString(result))
+      val paymentDateRow = document.select(".govuk-summary-list__row").asScala.toList(0)
+      assertRow(paymentDateRow, "Dyddiad talu", "Heddiw", Some("Newid"), Some("some-link-to-pay-frontend"))
+    }
+
+    "[PtaSa] should render the payment date row correctly" in {
+      PayApiStub.stubForFindBySessionId2xx(TestJourneys.PtaSa.testPtaSaJourneyUpdatedWithRefAndAmount)
+      val result = systemUnderTest.renderPage(fakeRequest())
+      val document = Jsoup.parse(contentAsString(result))
+      val paymentDateRow = document.select(".govuk-summary-list__row").asScala.toList(0)
+      assertRow(paymentDateRow, "Payment date", "Today", Some("Change"), Some("some-link-to-pay-frontend"))
+    }
+
+    "[PtaSa] should render the payment date row correctly in Welsh" in {
+      PayApiStub.stubForFindBySessionId2xx(TestJourneys.PtaSa.testPtaSaJourneyUpdatedWithRefAndAmount)
+      val result = systemUnderTest.renderPage(fakeRequestWelsh())
+      val document = Jsoup.parse(contentAsString(result))
+      val paymentDateRow = document.select(".govuk-summary-list__row").asScala.toList(0)
+      assertRow(paymentDateRow, "Dyddiad talu", "Heddiw", Some("Newid"), Some("some-link-to-pay-frontend"))
     }
 
   }
 
-  private def assertRow(element: Element, keyText: String, valueText: String, actionText: String, actionHref: String): Assertion = {
+  private def assertRow(element: Element, keyText: String, valueText: String, actionText: Option[String], actionHref: Option[String]): Assertion = {
     element.select(".govuk-summary-list__key").text() shouldBe keyText
     element.select(".govuk-summary-list__value").text() shouldBe valueText
-    element.select(".govuk-summary-list__actions").text() shouldBe actionText
-    element.select(".govuk-summary-list__actions").select("a").attr("href") shouldBe actionHref
+    actionText.fold(element.select(".govuk-summary-list__actions").text() shouldBe "")(content => element.select(".govuk-summary-list__actions").text() shouldBe content)
```

Comment:
Just realised we should probably do a slightly different assertion for the change link not being there, I'll pair with you in the morning